autosklearn/pipeline/components/data_preprocessing/feature_reduction/truncated_svd.py

```diff
@@ -0,0 +1,77 @@
+from typing import Dict, Optional, Tuple, Union
+
+from ConfigSpace.configuration_space import ConfigurationSpace
+import ConfigSpace.hyperparameters as CSH
+
+import numpy as np
+from scipy.sparse import csr_matrix
+
+from autosklearn.pipeline.base import DATASET_PROPERTIES_TYPE, PIPELINE_DATA_DTYPE
+from autosklearn.pipeline.components.base import AutoSklearnPreprocessingAlgorithm
+from autosklearn.pipeline.constants import DENSE, SPARSE, UNSIGNED_DATA, INPUT
+
+from sklearn.decomposition import TruncatedSVD
+
+
+class FeatureReduction(AutoSklearnPreprocessingAlgorithm):
+    """
+    Reduces the features created by a bag of words encoding
+    """
+
+    def __init__(
+            self,
+            n_components=None,
+            random_state: Optional[Union[int, np.random.RandomState]] = None
+    ) -> None:
+        self.n_components = n_components
+        self.random_state = random_state
+
+    def fit(self, X: PIPELINE_DATA_DTYPE, y: Optional[PIPELINE_DATA_DTYPE] = None
+            ) -> 'FeatureReduction':
+        X = csr_matrix(X)
```

Comment:
i do not remember what i excellently thought about, but i think it was because in the truncedsvd documentation is mentioned that one of the main advantages of  truncedsvd in comparison to pca is that it can deal better with sparse matrices but i think you are right and it does not make so much sense here.

Reply:
Hmm, that is true but I'm not sure it requires an explicit sparse matrix type, more generally that the method itself works on matrices with sparse entries. I would hesitate to convert the matrix to sparse unless we can guarantee that it is actually sparse before converting it.

@mfeurer thoughts?